httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -499,6 +499,18 @@ public URIBuilder setPath(final String path) {
         return this;
     }
 
+    /**
+     * Appends path to URI. The value is expected to be unescaped and may contain non ASCII characters.
+     *
+     * @return this.
+     */
+    public URIBuilder appendPath(final String path) {
+        final List<String> segments = new ArrayList<>(getPathSegments());
+        segments.addAll(path != null ? splitPath(path) : new ArrayList<String>());
+        setPathSegments(segments);
```

Comment:
@larshelge Should not it be possible to re-use `#setPathSegments` here?

Reply:
I fully agree. I thought about it but favored consistency with existing code, e.g. `setPathSegments(String...)` and `setPathSegments(List)`. I am happy to change it.

Reply:
Sorry, I did not fully understand - do you mean re-use `appendPathSegments(List)` ?

Reply:
> Sorry, I did not fully understand - do you mean re-use `appendPathSegments(List)` ?

@larshelge Yes, that is what I meant. I thought it would make it a bit neater. Not very important, though.

Reply:
@ok2c Got it. I will update it.

Reply:
@ok2c Made the changes now.